wchar/wchar.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * wchar functions
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Damian Loewnau
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+/* TODO: add implementations for remaining wchar functions */
+#include <wchar.h>
+
+
+int wcscmp(const wchar_t *ws1, const wchar_t *ws2)
+{
+	int i, ret = 1;
+
+	for (i = 0; ws1[i] == ws2[i]; i++) {
+		if (ws1[i] == L'\0') {
+			ret = 0;
+			break;
+		}
+	}
+
+	if (ret) {
+		ret = ws1[i] - ws2[i];
+	}
```

Comment:
Two MISRA problems here:
- if (ret) -> if (ret != 0)
- ret = ws1[i] - ws2[i]; is an implicite cast wchar_t -> int. This a cast between different essential types, so it's not allowed. Subtraction is an efficient approach, but simple comparision mitigates mentioned problem

Reply:
Thanks, do You mean sth like changes in last force-push?